R/profilePlot.R

```diff
@@ -112,209 +105,322 @@ profilePlot <- function(contrastDF,
+                                                   color = referenceLine,
+                                                   width = refLineThickness,
+                                                   y     = 0)
         }
-
         if (!is.null(foldChangeLines)) {
-            decorations <- getCxPlotDecorations(decorations = decorations,
-                                                color       = symbolFill[which(groupNames == "Increased")],
-                                                width       = refLineThickness,
-                                                y           = foldChangeLines)
-            decorations <- getCxPlotDecorations(decorations = decorations,
-                                                color       = symbolFill[which(groupNames == "Decreased")],
-                                                width       = refLineThickness,
-                                                y           = -1*foldChangeLines)
+            decorations <- .getCxPlotDecorations(decorations = decorations,
+                                                 color       = colors[2],
+                                                 width       = refLineThickness,
+                                                 y           = foldChangeLines)
+            decorations <- .getCxPlotDecorations(decorations = decorations,
+                                                 color       = colors[1],
+                                                 width       = refLineThickness,
+                                                 y           = -1 * foldChangeLines)
+        }
+        events <- htmlwidgets::JS("{ 'mousemove' : function(o, e, t) {
+                                                if (o != null && o != false) {
+                                                  if (o.y != null &&
+                                                      o.y.data != null &&
+                                                      o.y.smps != null) {
+                                                      info = '<b>' + o.y.vars[0]  + '</b>' + '<br/>' +
+                                                             '<b>' + o.y.smps[0]  + '</b>' + ': ' + o.y.data[0][0] + '<br/>' +
+                                                             '<b>' + o.y.smps[1]  + '</b>' + ': ' + o.y.data[0][1] ;
+                                                      if (o.z != null && o.z['rgd_symbol'] != null) {
+                                                        info  = info + '<br/>' +
+                                                              '<b> Symbol</b>' + ': ' + o.z['rgd_symbol'] ;
+                                                      }
+                                                    t.showInfoSpan(e, info);
+
+                                                  }
+                                                }; }}")
+        if (missing(geneSymCol) && sizeBySignificance) {
```

Comment:
Do you need 4 if loops. Why cant we create the var.annot data based on the individual flags and not all at one time.

Reply:
* There are no loops here, only if statements.
* I did not fully understand the question/suggestion as I already building var.annot based on the flags in the ifs.

Reply:
My bad. I didnt mean a loop. But i meant the if statements. Can you remodify it something as below. We can reduce it to 2 if statements rather than 4. Sample code below(not tested)

if (sizeBySignificance){
var.annot <- contrastDF %>% dplyr::select(group,negLog10P) 
 sizeBy <- "negLog10P"
 showSizeLegend <- TRUE
} else {
contrastDF %>%
                dplyr::select(Group)
 sizeBy <- "Group"
 showSizeLegend <- FALSE
}

if (!missing(geneSymCol) {
var.annot <- left_join(var.annot, contrastDF %>% select(geneSymcol), by = "<<rownames>>"
}
}